external-crates/move/crates/move-package/src/resolution/dependency_graph.rs

```diff
@@ -447,10 +484,18 @@ impl<Progress: Write> DependencyGraphBuilder<Progress> {
                     &dep_pkg_path,
                     &mut self.progress_output,
                 )?;
-                (pkg_graph, false, true, dep_pkg_name)
+                // TODO: support resolved_pkg_name and resolved_version for
```

Comment:
The issue here is that both the resolve name and version hooks derive from the dependency's manifest file instead of lockfile, but with externally resolved deps we effectively have just the lockfile. Since external deps aren't really used on Sui (and I wasn't sure whether this is something vestigial that would be deprecated), I left this as is for this PR to keep the scope sane.
We can solve this by adding more info to the lockfile (resolved name and version) and / or making the hooks also be able to work with lockfiles instead of just manifest. Although if externally resolved deps aren't going to be used on Sui, perhaps it's not worth the effort?

Reply:
I would be open to adding more info to the lock file to support this.  For background the external resolution logic was initially intended to support a package manager -- the original package manager that it was intended to support (Movey) did not materialise, but we do expect others to take its place, and there are some plans to use external resolution as a way to simplify how the framework dependencies interact with the Move toolchain, so I don't think external resolution will disappear.

Reply:
Cool, in that case I can add it in a follow up PR before we enable the hooks (there's no difference before then).

With this PR in the lockfile we now have:
- resolved name for each dep (the resolved name is now stored in the name field where the original name is stored without the hook; not for the root package -- we resolve it from the manifest)
- resolved version for each dep (not for the root package -- we resolve it from the manifest)

We don't have:
(a) original name for each dep (this is needed for user friendly error messages; for root we can always get it from the manifest)
(b) resolved name for the root package (this is actually needed only for the external resolver because otherwise we can resolve it from the dep manifest)
(c) version for the root package (also only needed for the external resolver for the same reason).

So the two obvious options I can see for adding this are:
(1) add (a), (b), and (c) to the lockfile
(2) add only (a) to the lockfile while for (b) and (c) we can add it as some extra metadata just for the external resolver

I'm more keen on (1) because with (b) and (c) we effectively get the original id and published-at into the lockfile which means that, with a reasonable tooling tweak, we can completely move them out of the manifest.

Reply:
Agreed, option (1) seems reasonable, and in fact we have plans to introduce the package's published-at and original-id into the lock file anyway (to support automated address management and address domain separation).

Reply:
👆 what @amnn said, we have plans to add related things here. Though one thing to keep in mind though is that we plan to namespace this based on network, so there won't be only 3 three fields, but one for each network (with devnet rotating).

> I can add it in a follow up PR before we enable the hooks (there's no difference before then)

So maybe we can compare notes / share plans here offline before making changes.

In general, the (a)/(b)/(c) breakdown is super useful and I'm in favor of being more explicit about info (option (1)) than not.